### PR TITLE
Repair Next.js Edge SDK

### PR DESCRIPTION
**Pull request by deltaepsilon**
## Summary

Edge doesn't provide `global`, so we're using `AsyncLocalStorage` instead.  

Testing also revealed that the header parsing logic was inverted, causing headers to get lost on all Edge requests. 

## How did you test this change?

Deploys to Vercel

## Are there any deployment considerations?

No.

## Does this work require review from our design team?

Batch bump to Next SDK

### CODE REVIEW

File: docs-content/getting-started/fullstack-frameworks/next-js/2_page-router.md [hunk 1]
```diff
@@ -139,11 +139,52 @@ function ThrowerOfErrors({
 }
 ```
 
-## Catch server-side render (SSR) errors (optional)
+## Enable server-side tracing
```
Comment (deltaepsilon) [inline docs-content/getting-started/fullstack-frameworks/next-js/2_page-router.md:142]: It turns out that `instrumentation.ts` captures traces, but only in certain cases can it capture SSR errors. So `_error.tsx` and `error.tsx` are still the most reliable ways to capture SSR errors.

In short, `instrumentation.ts` doesn't currently catch Edge SSR. Our `registerHighlight` handler is currently empty in Edge environments, but OpenTelemetry doesn't really work in Edge anyway, so I don't suspect that Next.js instruments Edge routes anyway.

File: e2e/nextjs/instrumentation.ts [hunk 2]
```diff
@@ -2,13 +2,11 @@
 import { CONSTANTS } from '@/constants'
 
 export async function register() {
-	if (process.env.NEXT_RUNTIME === 'nodejs') {
```
Comment (deltaepsilon) [inline e2e/nextjs/instrumentation.ts:5]: This check is no longer necessary due to better SDK structure.

File: e2e/nextjs/pages/redirect.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ type Props = {
 }
 export default function RedirectPage({ shouldRedirect }: Props) {
 	if (shouldRedirect) {
-		return redirect(`/isr`)
+		return redirect(`/ssr`)
```
Comment (deltaepsilon) [inline e2e/nextjs/pages/redirect.tsx:8]: `/isr` was just confusing. Switching to `/ssr` across the board.

File: sdk/highlight-next/src/util/highlight-edge.ts [hunk 4]
```diff
@@ -21,6 +22,7 @@ let globalRequestMetadata: RequestMetadata = {
 	requestId: '',
 }
 let executionContext: ExtendedExecutionContext
+const asyncLocalStorage = new AsyncLocalStorage<HighlightContext>()
```
Comment (deltaepsilon) [inline sdk/highlight-next/src/util/highlight-edge.ts:25]: Copy/paste from the `@highlight-run/node` implementation. It worked first try.

File: sdk/highlight-next/src/util/types.ts [hunk 5]
```diff
@@ -71,4 +71,4 @@ export interface HighlightInterface {
 	flush: () => Promise<void>
 }
 
-export const HIGHLIGHT_REQUEST_HEADER = 'X-Highlight-Request'
+export const HIGHLIGHT_REQUEST_HEADER = 'x-highlight-request'
```
Comment (deltaepsilon) [inline sdk/highlight-next/src/util/types.ts:74]: I'm not sure if capitalization hurt here, but it definitely didn't help!

File: sdk/highlight-next/src/util/with-highlight-edge.ts [hunk 6]
```diff
@@ -19,7 +19,7 @@ export function Highlight(env: HighlightEnv) {
 		) {
 			H.initEdge(request, env, event)
 			const headers: IncomingHttpHeaders = {}
-			request.headers.forEach((k, v) => (headers[k] = v))
+			request.headers.forEach((k, v) => (headers[v] = k))
```
Comment (deltaepsilon) [inline sdk/highlight-next/src/util/with-highlight-edge.ts]: This was baffling. IT took me a while to realize that the keys and values were flipped!

Comment (deltaepsilon) [reply sdk/highlight-next/src/util/with-highlight-edge.ts]: See https://developer.mozilla.org/en-US/docs/Web/API/Headers/forEach

File: sdk/highlight-next/src/util/with-highlight-edge.ts [hunk 7]
```diff
@@ -19,7 +19,7 @@ export function Highlight(env: HighlightEnv) {
 		) {
 			H.initEdge(request, env, event)
 			const headers: IncomingHttpHeaders = {}
-			request.headers.forEach((k, v) => (headers[k] = v))
+			request.headers.forEach((v, k) => (headers[k] = v))
```
Comment (deltaepsilon) [inline sdk/highlight-next/src/util/with-highlight-edge.ts:22]: This was baffling at first, but MDN explains the flipped args:

https://developer.mozilla.org/en-US/docs/Web/API/Headers/forEach

**Review by Vadman97 — APPROVED** [review]

**Pull request merged**